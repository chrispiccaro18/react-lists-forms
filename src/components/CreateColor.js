import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

export default class CreateColor extends PureComponent {
  static propTypes = {
    addColor: PropTypes.func.isRequired
  }

  state = {
    colorName: '',
    color: '#FF0000'
  }

  handleSubmit = event => {
    event.preventDefault();
    const { colorName, color } = this.state;
    this.props.addColor({ colorName, hex: color });
  }

  handleChange = ({ target }) => {
    this.setState({ [target.name]: target.value });
  }

  render() {
    const { colorName, color } = this.state;
    return (
      <>
        <form onSubmit={this.handleSubmit}>
          <input type="text" name="colorName" value={colorName} onChange={this.handleChange} />
          <input type="color" name="color" value={color} onChange={this.handleChange} />
          <button>Create Color</button>
        </form>
      </>
    );
  }
}

