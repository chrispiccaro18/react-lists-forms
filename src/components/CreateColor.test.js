import React from 'react';
import { shallow } from 'enzyme';
import CreateColor from './CreateColor';

describe('CreateColor component', () => {
  const mockAddColor = jest.fn(x => 42 + x);
  
  it('renders a CreateColor', () => {
    const wrapper = shallow(<CreateColor addColor={mockAddColor} />);
    expect(wrapper).toMatchSnapshot();
  });
});
