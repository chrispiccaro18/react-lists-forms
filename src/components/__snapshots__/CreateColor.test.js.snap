// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateColor component renders a CreateColor 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      name="colorName"
      onChange={[Function]}
      type="text"
      value=""
    />
    <input
      name="color"
      onChange={[Function]}
      type="color"
      value="#ff0000"
    />
    <button>
      Create Color
    </button>
  </form>
</Fragment>
`;
